src/krb5/_principal.pyx

```diff
@@ -186,3 +392,47 @@ def unparse_name_flags(
         return <bytes>name
     finally:
         krb5_free_unparsed_name_generic(context.raw, name)
+
+
+# Note: build_principal() does not actually call krb5_build_principal() because
```

Comment:
What's the purpose of `krb5_build_principal`. I'm just wondering if this is worth doing over just exposing `krb5_build_principal_alloc_va` for MIT only and hope in the future Heimdal includes it.

Reply:
Makes sense, Cython should be able to build an array of null terminated strings, probably would require some memory copies into a custom buffer. I suppose in the end we can include this as is and as the tests are working then no harm done. Pity Cython doesn't have a nice way of doing varargs.